device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -379,9 +584,34 @@ def get_eeprom_dict(self, port_num):
                 sfp_data['interface'] = sfpi_obj.get_data_pretty()
 
                 # check if it is a 100G module
-                if sfp_data['interface']['data']['Identifier'] == 'QSFP28 or later':
+                if sfp_data['interface']['data']['Identifier'] not in [type_of_transceiver['18'], type_of_transceiver['19']]:
                     return self.get_qsfp_data(eeprom_ifraw)
 
+                # decode application advertisement
```

Comment:
Who is using these new attributes?

Reply:
This is necessary for CMIS3/4 init sequence, for now, we blindly select application 2, but it's not always correct, so I added the support of decoding application advertisement, in the case that we need to add more module-specific init sequence in the future.
Note: The changes in this PR is intentionally revised to InnoLight T-DP4CNT-N00 only, the common CMIS3/4 support is not in our scope.